gcp/appengine/utils.py

```diff
@@ -23,3 +27,43 @@ def is_prod():
   # and it's only used for Redis cache (which has its own env vars) and logging.
   # Consider removing this altogether.
   return 'GAE_ENV' in os.environ or 'K_SERVICE' in os.environ
+
+
+def relative_time(value: datetime | str) -> str:
+  """
+    Convert a datetime or ISO 8601 string to a human-readable relative time.
+
+    Args:
+        value: The input datetime as an ISO 8601 string or a datetime object.
+
+    Returns:
+        str: A human-readable string representing the relative time as:
+             - < 1 minute: Just now
+             - < 1 hour: x minutes ago
+             - < 1 day: x days ago
+             - < 2 day: yesterday
+             - < 7 days: x days ago
+             - >= 7 days: DD MMM
+             - Year is not current year: DD MMM YYYY
+    """
+  # Parse the ISO 8601 string to a datetime object
+  if isinstance(value, str):
+    value = isoparse(value)
+
+  now = datetime.now(tz=tz.tzutc())
+  diff = relativedelta(now, value)
+  diff_seconds = (now - value).total_seconds()
+
+  if diff_seconds < 60:
+    return "just now"
+  if diff_seconds < 3600:
+    return f"{diff.minutes} minute{'s' if diff.minutes > 1 else ''} ago"
+  if diff_seconds < 86400:
+    return f"{diff.hours} hour{'s' if diff.hours > 1 else ''} ago"
+  if diff_seconds < 172800:
+    return "yesterday"
+  if diff_seconds < 604800:
+    return f"{diff.days} day{'s' if diff.days > 1 else ''} ago"
```

Comment:
An alternative to having these precalculated numbers here is to just match by whether the next unit is zero: i.e.
```suggestion
  if diff_minutes == 0:
    return "just now"
  if diff_hours == 0:
    return f"{diff.minutes} minute{'s' if diff.minutes > 1 else ''} ago"
  if diff.days == 0:
    return f"{diff.hours} hour{'s' if diff.hours > 1 else ''} ago"
  if diff.days == 1:
    return "yesterday"
  if diff.days < 7:
    return f"{diff.days} day{'s' if diff.days > 1 else ''} ago"
```

(Though you might want to double check this, I'm not sure if this will work)

Reply:
Nice! Yes it works!